Wyam.Core/Modules/Sitemap.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Wyam.Common.Documents;
+using Wyam.Common.IO;
+using Wyam.Common.Modules;
+using Wyam.Common.Pipelines;
+using Wyam.Core.Documents;
+
+namespace Wyam.Core.Modules
+{
+    public class Sitemap : IModule
+    {
+        private string sitemapFilename;
+        private Func<string, string> locationFormatter;
+        private static readonly string[] ChangeFrequencies = new string[] { "always", "hourly", "daily", "weekly", "monthly", "yearly", "never" };
+
+        public Sitemap(string sitemapFilename, Func<string, string> locationFormatter = null)
+        {
+            this.sitemapFilename = sitemapFilename;
+            this.locationFormatter = locationFormatter;
+        }
+
+        public IEnumerable<IDocument> Execute(IReadOnlyList<IDocument> inputs, IExecutionContext context)
+        {
+            var outputFilename = Path.Combine(context.OutputFolder, this.sitemapFilename);
+            var docs = context.Documents.Where(f => f.Metadata.ContainsKey(MetadataKeys.SitemapItem)).ToArray();
+
+            if (docs.Length > 0)
+            {
+                var sb = new StringBuilder();
+                sb.Append("<?xml version=\"1.0\" encoding=\"UTF-8\"?><urlset xmlns=\"http://www.sitemaps.org/schemas/sitemap/0.9\">");
+
+                foreach (var doc in docs)
+                {
+                    object hostname = null;
+                    doc.Metadata.TryGetValue(MetadataKeys.Hostname, out hostname);
+
+                    var smi = doc.Metadata.Get(MetadataKeys.SitemapItem);
+                    var itm = smi as SitemapItem;
+
+                    if (itm == null)
+                        itm = new SitemapItem(smi.ToString());
```

Comment:
That would only work when using your suggested inputs method using `Documents` before `Sitemap`. Otherwise all documents from all pipelines that have a value for `RelativeFilePath` would be added to the sitemap.

Allowing to omit the `SitemapItem` meta data would make it impossible to exclude some urls from a pipeline. In my example project I used the `If` module to omit the `SitemapItem` meta data from specific urls. This wouldn't be possible when we would use only `RelativeFilePath`. Do you agree on this?